libpdb/subscription.go

```diff
@@ -0,0 +1,150 @@
+package libpdb
+
+import (
+	"bytes"
+	"encoding/binary"
+	"encoding/gob"
+	"errors"
+	"github.com/ryscheng/pdb/common"
+	"github.com/ryscheng/pdb/drbg"
+	"golang.org/x/crypto/nacl/box"
+  "github.com/agl/ed25519"
+)
+
+type Subscription struct {
+	// for random looking pir requests
+	drbg *drbg.HashDrbg
```

Comment:
Seems like the right thing to do is to have a common TopicHandle between publishers and subscribers. No reason to flatten out state separately for publishers and subscribers right?